# Issue 4576: biopython optional package upgrade to 1.49beta

archive/issues_004576.json:
```json
{
    "assignees": [
        "https://github.com/sagetrac-mabshoff"
    ],
    "body": "Biopython 1.49 beta was released November 7th, 2008.  Its a bigger release than the number indicates - one nice thing was a change from requiring Numeric for some things to using numpy consistently, which makes the install simpler and things are more sage-compatible.\n\nThe install should be tested on something other than what I have available right now which is an intel mac running 10.5.\n\nThe package is at: \n[http://www.d.umn.edu/~mhampton/biopython-1.49b.spkg](http://www.d.umn.edu/~mhampton/biopython-1.49b.spkg)\n\nComponent: **packages: optional**\n\nKeywords: **biopython**\n\n_Issue created by migration from https://trac.sagemath.org/ticket/4576_\n\n",
    "closed_at": "2008-11-29T21:43:14Z",
    "created_at": "2008-11-21T14:58:48Z",
    "labels": [
        "https://github.com/sagemath/sage/labels/c%3A%20packages%3A%20optional",
        "https://github.com/sagemath/sage/labels/p%3A%20minor%20/%204",
        "https://github.com/sagemath/sage/labels/enhancement",
        "https://github.com/sagemath/sage/labels/c%3A%20packages%3A%20optional"
    ],
    "milestone": "https://github.com/sagemath/sage/milestones/sage-3.2.1",
    "reactions": [],
    "repository": "https://github.com/sagemath/sage",
    "title": "biopython optional package upgrade to 1.49beta",
    "type": "issue",
    "updated_at": "2008-11-29T21:43:14Z",
    "url": "https://github.com/sagemath/sage/issues/4576",
    "user": "https://github.com/sagetrac-mhampton"
}
```
Biopython 1.49 beta was released November 7th, 2008.  Its a bigger release than the number indicates - one nice thing was a change from requiring Numeric for some things to using numpy consistently, which makes the install simpler and things are more sage-compatible.

The install should be tested on something other than what I have available right now which is an intel mac running 10.5.

The package is at: 
[http://www.d.umn.edu/~mhampton/biopython-1.49b.spkg](http://www.d.umn.edu/~mhampton/biopython-1.49b.spkg)

Component: **packages: optional**

Keywords: **biopython**

_Issue created by migration from https://trac.sagemath.org/ticket/4576_





---

archive/issue_events_051797.json:
```json
{
    "actor": "https://github.com/sagetrac-mhampton",
    "created_at": "2008-11-21T14:58:48Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "milestone_number": null,
    "milestone_title": "sage-3.2.2",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51797"
}
```



---

archive/issue_events_051798.json:
```json
{
    "actor": "https://github.com/sagetrac-mhampton",
    "created_at": "2008-11-21T14:58:48Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "label": "https://github.com/sagemath/sage/labels/c%3A%20packages%3A%20optional",
    "label_color": "0000b0",
    "label_name": "c: packages: optional",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51798"
}
```



---

archive/issue_events_051799.json:
```json
{
    "actor": "https://github.com/sagetrac-mhampton",
    "created_at": "2008-11-21T14:58:48Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "label": "https://github.com/sagemath/sage/labels/p%3A%20minor%20/%204",
    "label_color": "ffe799",
    "label_name": "p: minor / 4",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51799"
}
```



---

archive/issue_events_051800.json:
```json
{
    "actor": "https://github.com/sagetrac-mhampton",
    "created_at": "2008-11-21T14:58:48Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "label": "https://github.com/sagemath/sage/labels/enhancement",
    "label_color": "696969",
    "label_name": "enhancement",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51800"
}
```



---

archive/issue_events_051801.json:
```json
{
    "actor": "https://github.com/sagetrac-mhampton",
    "created_at": "2008-11-21T14:58:48Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "label": "https://github.com/sagemath/sage/labels/c%3A%20packages%3A%20optional",
    "label_color": "0000b0",
    "label_name": "c: packages: optional",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51801"
}
```



---

archive/issue_events_051802.json:
```json
{
    "actor": "https://github.com/sagetrac-mabshoff",
    "created_at": "2008-11-21T14:58:48Z",
    "event": "assigned",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "subject": "https://github.com/sagetrac-mhampton",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51802"
}
```



---

archive/issue_comments_027173.json:
```json
{
    "body": "<div id=\"comment:1\" align=\"right\">comment:1</div>\n\nThis builds and installs fine on my Ubuntu 8.10 machine.",
    "created_at": "2008-11-21T17:05:17Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/4576#issuecomment-27173",
    "user": "https://github.com/mwhansen"
}
```

<div id="comment:1" align="right">comment:1</div>

This builds and installs fine on my Ubuntu 8.10 machine.



---

archive/issue_comments_027174.json:
```json
{
    "body": "<div id=\"comment:2\" align=\"right\">comment:2</div>\n\nThe tarball has a massive amount of meta-crap:\n\n```\n...\nbiopython-1.49b/biopython-1.49b/Tests/MEME/._mast.dna.oops.txt\nbiopython-1.49b/biopython-1.49b/Tests/MEME/mast.dna.oops.txt\nbiopython-1.49b/biopython-1.49b/Tests/MEME/._mast.protein.oops.txt\nbiopython-1.49b/biopython-1.49b/Tests/MEME/mast.protein.oops.txt\nbiopython-1.49b/biopython-1.49b/Tests/MEME/._mast.protein.tcm.txt\nbiopython-1.49b/biopython-1.49b/Tests/MEME/mast.protein.tcm.txt\nbiopython-1.49b/biopython-1.49b/Tests/MEME/._meme.dna.oops.txt\nbiopython-1.49b/biopython-1.49b/Tests/MEME/meme.dna.oops.txt\nbiopython-1.49b/biopython-1.49b/Tests/MEME/._meme.protein.oops.txt\nbiopython-1.49b/biopython-1.49b/Tests/MEME/meme.protein.oops.txt\nbiopython-1.49b/biopython-1.49b/Tests/MEME/._meme.protein.tcm.txt\nbiopython-1.49b/biopython-1.49b/Tests/MEME/meme.protein.tcm.txt\n```\n\nNote all the ._ files.  OS X sticks them there.  Can you make the \nthing again using sage.math (or any other linux box)?\n\nThe install works fine on sage.math:\n\n```\nreal\t0m21.822s\nuser\t0m17.317s\nsys\t0m2.876s\nSuccessfully installed biopython-1.49b\n```\n\nHow do I test that it works?  Can you give a few sample commands or explain how to run the biopython test suite?\n\nWilliam",
    "created_at": "2008-11-21T22:26:18Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/4576#issuecomment-27174",
    "user": "https://github.com/williamstein"
}
```

<div id="comment:2" align="right">comment:2</div>

The tarball has a massive amount of meta-crap:

```
...
biopython-1.49b/biopython-1.49b/Tests/MEME/._mast.dna.oops.txt
biopython-1.49b/biopython-1.49b/Tests/MEME/mast.dna.oops.txt
biopython-1.49b/biopython-1.49b/Tests/MEME/._mast.protein.oops.txt
biopython-1.49b/biopython-1.49b/Tests/MEME/mast.protein.oops.txt
biopython-1.49b/biopython-1.49b/Tests/MEME/._mast.protein.tcm.txt
biopython-1.49b/biopython-1.49b/Tests/MEME/mast.protein.tcm.txt
biopython-1.49b/biopython-1.49b/Tests/MEME/._meme.dna.oops.txt
biopython-1.49b/biopython-1.49b/Tests/MEME/meme.dna.oops.txt
biopython-1.49b/biopython-1.49b/Tests/MEME/._meme.protein.oops.txt
biopython-1.49b/biopython-1.49b/Tests/MEME/meme.protein.oops.txt
biopython-1.49b/biopython-1.49b/Tests/MEME/._meme.protein.tcm.txt
biopython-1.49b/biopython-1.49b/Tests/MEME/meme.protein.tcm.txt
```

Note all the ._ files.  OS X sticks them there.  Can you make the 
thing again using sage.math (or any other linux box)?

The install works fine on sage.math:

```
real	0m21.822s
user	0m17.317s
sys	0m2.876s
Successfully installed biopython-1.49b
```

How do I test that it works?  Can you give a few sample commands or explain how to run the biopython test suite?

William



---

archive/issue_events_051803.json:
```json
{
    "actor": "https://github.com/williamstein",
    "created_at": "2008-11-21T22:26:27Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "label": "https://github.com/sagemath/sage/labels/needs%20work",
    "label_color": "ffff00",
    "label_name": "needs work",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51803"
}
```



---

archive/issue_comments_027175.json:
```json
{
    "body": "<div id=\"comment:4\" align=\"right\">comment:4</div>\n\nI don't have an account on sage.math, maybe its time to get one.  Right now I'm in Thunder Bay, Ontario, so my options are more limited than usual.  I could also try to write something remove all the cruft - perhaps I could delete every file starting with \"._\"?  \n\nThe distributed biopython test suite is in the Tests folder of the spkg, I think it is not installed currently into sage.  To run it you just do \"python run_tests.py\" within the biopython/Tests directory.  \n\nHere are some simple things to do:\n\n```\nSearch for nucleotide records of Salvia divinorum (sage of the seers):\nfrom Bio import Entrez, SeqIO\nEntrez.email = \"hamptonio@gmail.com\" # Always tell NCBI who you are \nhandle = Entrez.esearch(db=\"Nucleotide\", term=\"salvia divinorum\") \nrecord = Entrez.read(handle)\nrecord[\"IdList\"]\n['113715469', '113715388', '111119918', '81230846', '81230845',\n'46326321', '1695982', '349139']\n```\n\nTake the first record found above and get its GenBank file, parse it, and read the description:\n\n```\nfrom Bio import Entrez, SeqIO\nEntrez.email = \"hamptonio@gmail.com\" # Always tell NCBI who you are \nhandle = Entrez.efetch(db=\"nucleotide\", id=\"113715469\", rettype=\"genbank\") \nrec = SeqIO.parse(handle,'genbank')\nx = rec.next()\nx.description\n'Salvia divinorum isolate x093 tRNA-Leu (trnL) gene and trnL-trnF\nintergenic spacer, partial sequence; chloroplast.'\n```\n\nor check its taxononomy:\n\n```\nx.annotations['taxonomy']\n['Eukaryota', 'Viridiplantae', 'Streptophyta', 'Embryophyta',\n'Tracheophyta', 'Spermatophyta', 'Magnoliophyta', 'eudicotyledons',\n'core eudicotyledons', 'asterids', 'lamiids', 'Lamiales', 'Lamiaceae',\n'Nepetoideae', 'Mentheae', 'Salvia']\n```",
    "created_at": "2008-11-22T00:33:25Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/4576#issuecomment-27175",
    "user": "https://github.com/sagetrac-mhampton"
}
```

<div id="comment:4" align="right">comment:4</div>

I don't have an account on sage.math, maybe its time to get one.  Right now I'm in Thunder Bay, Ontario, so my options are more limited than usual.  I could also try to write something remove all the cruft - perhaps I could delete every file starting with "._"?  

The distributed biopython test suite is in the Tests folder of the spkg, I think it is not installed currently into sage.  To run it you just do "python run_tests.py" within the biopython/Tests directory.  

Here are some simple things to do:

```
Search for nucleotide records of Salvia divinorum (sage of the seers):
from Bio import Entrez, SeqIO
Entrez.email = "hamptonio@gmail.com" # Always tell NCBI who you are 
handle = Entrez.esearch(db="Nucleotide", term="salvia divinorum") 
record = Entrez.read(handle)
record["IdList"]
['113715469', '113715388', '111119918', '81230846', '81230845',
'46326321', '1695982', '349139']
```

Take the first record found above and get its GenBank file, parse it, and read the description:

```
from Bio import Entrez, SeqIO
Entrez.email = "hamptonio@gmail.com" # Always tell NCBI who you are 
handle = Entrez.efetch(db="nucleotide", id="113715469", rettype="genbank") 
rec = SeqIO.parse(handle,'genbank')
x = rec.next()
x.description
'Salvia divinorum isolate x093 tRNA-Leu (trnL) gene and trnL-trnF
intergenic spacer, partial sequence; chloroplast.'
```

or check its taxononomy:

```
x.annotations['taxonomy']
['Eukaryota', 'Viridiplantae', 'Streptophyta', 'Embryophyta',
'Tracheophyta', 'Spermatophyta', 'Magnoliophyta', 'eudicotyledons',
'core eudicotyledons', 'asterids', 'lamiids', 'Lamiales', 'Lamiaceae',
'Nepetoideae', 'Mentheae', 'Salvia']
```



---

archive/issue_events_051804.json:
```json
{
    "actor": "https://github.com/sagetrac-mhampton",
    "created_at": "2008-11-24T22:48:16Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "label": "https://github.com/sagemath/sage/labels/needs%20work",
    "label_color": "ffff00",
    "label_name": "needs work",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51804"
}
```



---

archive/issue_events_051805.json:
```json
{
    "actor": "https://github.com/sagetrac-mhampton",
    "created_at": "2008-11-24T22:48:16Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "label": "https://github.com/sagemath/sage/labels/needs%20review",
    "label_color": "7fff00",
    "label_name": "needs review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51805"
}
```



---

archive/issue_comments_027176.json:
```json
{
    "body": "<div id=\"comment:5\" align=\"right\">comment:5</div>\n\nI have made an spkg that I think is clean now.",
    "created_at": "2008-11-24T22:48:16Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/4576#issuecomment-27176",
    "user": "https://github.com/sagetrac-mhampton"
}
```

<div id="comment:5" align="right">comment:5</div>

I have made an spkg that I think is clean now.



---

archive/issue_events_051806.json:
```json
{
    "actor": "https://github.com/williamstein",
    "created_at": "2008-11-28T06:51:14Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "label": "https://github.com/sagemath/sage/labels/needs%20review",
    "label_color": "7fff00",
    "label_name": "needs review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51806"
}
```



---

archive/issue_events_051807.json:
```json
{
    "actor": "https://github.com/williamstein",
    "created_at": "2008-11-28T06:51:14Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "label": "https://github.com/sagemath/sage/labels/needs%20work",
    "label_color": "ffff00",
    "label_name": "needs work",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51807"
}
```



---

archive/issue_comments_027177.json:
```json
{
    "body": "<div id=\"comment:6\" align=\"right\">comment:6</div>\n\nREFEREE REPORT:\n\n1. The crap is gone. :-)\n\n2. I tried out some of the things above and it works.\n\n3. It would be good to have an spkg-check script, which would look like this. This would get run automatically when one does \n\n```\nexport SAGE_CHECK=\"yes\"\n```\nbefore installing. \nHere's spkg-check, probably:\n\n```\n#!/bin/sh\ncd biopython/Tests/\npython run_tests.py\n```\n\n4. However the test suite itself fails 5 tests.  Can you comment on this?\n\n```\nwas@sage:~/build/sage-3.2.1.alpha1/biopython-1.49b/biopython/Tests$ python run_tests.py\ntest_Ace ... ok\ntest_AlignIO ... ok\ntest_BioSQL ... skipping. Enter your settings in Tests/setup_BioSQL.py (not important if you do not plan to use BioSQL).\nok\ntest_BioSQL_SeqIO ... skipping. Enter your settings in Tests/setup_BioSQL.py (not important if you do not plan to use BioSQL).\nok\ntest_CAPS ... ERROR\ntest_Clustalw ... ok\ntest_Clustalw_tool ... skipping. Install clustalw or clustalw2 if you want to use Bio.Clustalw.\nok\ntest_Cluster ... FAIL\ntest_CodonTable ... ok\ntest_CodonUsage ... ok\ntest_Compass ... ok\ntest_Crystal ... ok\ntest_DocSQL ... skipping. Install MySQLdb if you want to use Bio.DocSQL.\nok\ntest_EmbossPrimer ... ok\ntest_Entrez ... ok\ntest_Enzyme ... ok\ntest_FSSP ... ok\ntest_Fasta ... ok\ntest_Fasta2 ... ok\ntest_File ... ok\ntest_GACrossover ... ok\ntest_GAMutation ... ok\ntest_GAOrganism ... ok\ntest_GAQueens ... ok\ntest_GARepair ... ok\ntest_GASelection ... ok\ntest_GFF ... skipping. Environment is not configured for this test (not important if you do not plan to use Bio.GFF).\nok\ntest_GFF2 ... skipping. Install MySQLdb if you want to use Bio.GFF.\nok\ntest_GenBank ... ok\ntest_GraphicsChromosome ... skipping. Install reportlab if you want to use Bio.Graphics.\nok\ntest_GraphicsDistribution ... skipping. Install reportlab if you want to use Bio.Graphics.\nok\ntest_GraphicsGeneral ... skipping. Install reportlab if you want to use Bio.Graphics.\nok\ntest_HMMCasino ... ok\ntest_HMMGeneral ... ok\ntest_HotRand ... ok\ntest_IsoelectricPoint ... ok\ntest_KDTree ... ERROR\ntest_KEGG ... ok\ntest_KeyWList ... ok\ntest_Location ... ok\ntest_LocationParser ... ok\ntest_LogisticRegression ... ok\ntest_MEME ... ok\ntest_MarkovModel ... ok\ntest_Medline ... ok\ntest_NCBIStandalone ... ok\ntest_NCBIXML ... ok\ntest_NCBI_qblast ... ok\ntest_NNExclusiveOr ... ok\ntest_NNGene ... ok\ntest_NNGeneral ... ok\ntest_Nexus ... ok\ntest_PDB ... ok\ntest_ParserSupport ... ok\ntest_Pathway ... ok\ntest_Phd ... ok\ntest_PopGen_FDist ... skipping. Install FDist if you want to use Bio.PopGen.FDist.\nok\ntest_PopGen_FDist_nodepend ... ok\ntest_PopGen_GenePop ... ok\ntest_PopGen_SimCoal ... skipping. Install SIMCOAL2 if you want to use Bio.PopGen.SimCoal.\nok\ntest_PopGen_SimCoal_nodepend ... ok\ntest_ProtParam ... ok\ntest_Registry ... ok\ntest_Restriction ... ERROR\ntest_SCOP_Astral ... ok\ntest_SCOP_Cla ... ok\ntest_SCOP_Des ... ok\ntest_SCOP_Dom ... ok\ntest_SCOP_Hie ... ok\ntest_SCOP_Raf ... ok\ntest_SCOP_Residues ... ok\ntest_SCOP_Scop ... ok\ntest_SProt ... ok\ntest_SVDSuperimposer ... ok\ntest_SeqIO ... `^Hok\ntest_SeqIO_online ... ok\ntest_SeqUtils ... ok\ntest_SubsMat ... ok\ntest_UniGene ... ok\ntest_Wise ... skipping. Install Wise2 (dnal) if you want to use Bio.Wise.\nok\ntest_align ... ok\ntest_docstrings ... ok\ntest_geo ... ok\ntest_interpro ... ok\ntest_kNN ... ok\ntest_pairwise2 ... ok\ntest_prodoc ... ok\ntest_property_manager ... ok\ntest_prosite ... ok\ntest_prosite2 ... ok\ntest_psw ... skipping. Install Wise2 (dnal) if you want to use Bio.Wise.\nok\ntest_seq ... ok\ntest_translate ... ok\ntest_trie ... ERROR\ntest_triefind ... ERROR\n\n======================================================================\nERROR: test_CAPS\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"run_tests.py\", line 125, in runTest\n    self.runSafeTest()\n  File \"run_tests.py\", line 138, in runSafeTest\n    cur_test = __import__(self.test_name)\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Tests/test_CAPS.py\", line 3, in <module>\n    from Bio.Restriction import *\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/Restriction/__init__.py\", line 61, in <module>\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/Restriction/Restriction.py\", line 96, in <module>\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/Restriction/PrintFormat.py\", line 14, in <module>\nImportError: No module named DNAUtils\n\n======================================================================\nERROR: test_KDTree\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"run_tests.py\", line 125, in runTest\n    self.runSafeTest()\n  File \"run_tests.py\", line 138, in runSafeTest\n    cur_test = __import__(self.test_name)\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Tests/test_KDTree.py\", line 1, in <module>\n    from Bio.KDTree.KDTree import _neighbor_test, _test\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/KDTree/__init__.py\", line 10, in <module>\n    #\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/KDTree/KDTree.py\", line 13, in <module>\nImportError: cannot import name _CKDTree\n\n======================================================================\nERROR: test_Restriction\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"run_tests.py\", line 125, in runTest\n    self.runSafeTest()\n  File \"run_tests.py\", line 138, in runSafeTest\n    cur_test = __import__(self.test_name)\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Tests/test_Restriction.py\", line 8, in <module>\n    from Bio.Restriction import *\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/Restriction/__init__.py\", line 61, in <module>\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/Restriction/Restriction.py\", line 96, in <module>\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/Restriction/PrintFormat.py\", line 13, in <module>\nImportError: cannot import name RanaConfig\n\n======================================================================\nERROR: test_trie\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"run_tests.py\", line 125, in runTest\n    self.runSafeTest()\n  File \"run_tests.py\", line 138, in runSafeTest\n    cur_test = __import__(self.test_name)\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Tests/test_trie.py\", line 6, in <module>\n    from Bio import trie\nImportError: cannot import name trie\n\n======================================================================\nERROR: test_triefind\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"run_tests.py\", line 125, in runTest\n    self.runSafeTest()\n  File \"run_tests.py\", line 138, in runSafeTest\n    cur_test = __import__(self.test_name)\n  File \"/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Tests/test_triefind.py\", line 6, in <module>\n    from Bio import trie\nImportError: cannot import name trie\n\n======================================================================\nFAIL: test_Cluster\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"run_tests.py\", line 125, in runTest\n    self.runSafeTest()\n  File \"run_tests.py\", line 162, in runSafeTest\n    expected_handle)\n  File \"run_tests.py\", line 263, in compare_output\n    % (repr(output_line), repr(expected_line))\nAssertionError: \nOutput  : 'test_clusterdistance (test_Cluster.TestCluster) ... ERROR\\n'\nExpected: 'test_clusterdistance (test_Cluster.TestCluster) ... ok\\n'\n\n----------------------------------------------------------------------\nRan 95 tests in 147.466s\n\nFAILED (failures=1, errors=5)\nYou have new mail in /var/mail/was\n```",
    "created_at": "2008-11-28T06:51:14Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/4576#issuecomment-27177",
    "user": "https://github.com/williamstein"
}
```

<div id="comment:6" align="right">comment:6</div>

REFEREE REPORT:

1. The crap is gone. :-)

2. I tried out some of the things above and it works.

3. It would be good to have an spkg-check script, which would look like this. This would get run automatically when one does 

```
export SAGE_CHECK="yes"
```
before installing. 
Here's spkg-check, probably:

```
#!/bin/sh
cd biopython/Tests/
python run_tests.py
```

4. However the test suite itself fails 5 tests.  Can you comment on this?

```
was@sage:~/build/sage-3.2.1.alpha1/biopython-1.49b/biopython/Tests$ python run_tests.py
test_Ace ... ok
test_AlignIO ... ok
test_BioSQL ... skipping. Enter your settings in Tests/setup_BioSQL.py (not important if you do not plan to use BioSQL).
ok
test_BioSQL_SeqIO ... skipping. Enter your settings in Tests/setup_BioSQL.py (not important if you do not plan to use BioSQL).
ok
test_CAPS ... ERROR
test_Clustalw ... ok
test_Clustalw_tool ... skipping. Install clustalw or clustalw2 if you want to use Bio.Clustalw.
ok
test_Cluster ... FAIL
test_CodonTable ... ok
test_CodonUsage ... ok
test_Compass ... ok
test_Crystal ... ok
test_DocSQL ... skipping. Install MySQLdb if you want to use Bio.DocSQL.
ok
test_EmbossPrimer ... ok
test_Entrez ... ok
test_Enzyme ... ok
test_FSSP ... ok
test_Fasta ... ok
test_Fasta2 ... ok
test_File ... ok
test_GACrossover ... ok
test_GAMutation ... ok
test_GAOrganism ... ok
test_GAQueens ... ok
test_GARepair ... ok
test_GASelection ... ok
test_GFF ... skipping. Environment is not configured for this test (not important if you do not plan to use Bio.GFF).
ok
test_GFF2 ... skipping. Install MySQLdb if you want to use Bio.GFF.
ok
test_GenBank ... ok
test_GraphicsChromosome ... skipping. Install reportlab if you want to use Bio.Graphics.
ok
test_GraphicsDistribution ... skipping. Install reportlab if you want to use Bio.Graphics.
ok
test_GraphicsGeneral ... skipping. Install reportlab if you want to use Bio.Graphics.
ok
test_HMMCasino ... ok
test_HMMGeneral ... ok
test_HotRand ... ok
test_IsoelectricPoint ... ok
test_KDTree ... ERROR
test_KEGG ... ok
test_KeyWList ... ok
test_Location ... ok
test_LocationParser ... ok
test_LogisticRegression ... ok
test_MEME ... ok
test_MarkovModel ... ok
test_Medline ... ok
test_NCBIStandalone ... ok
test_NCBIXML ... ok
test_NCBI_qblast ... ok
test_NNExclusiveOr ... ok
test_NNGene ... ok
test_NNGeneral ... ok
test_Nexus ... ok
test_PDB ... ok
test_ParserSupport ... ok
test_Pathway ... ok
test_Phd ... ok
test_PopGen_FDist ... skipping. Install FDist if you want to use Bio.PopGen.FDist.
ok
test_PopGen_FDist_nodepend ... ok
test_PopGen_GenePop ... ok
test_PopGen_SimCoal ... skipping. Install SIMCOAL2 if you want to use Bio.PopGen.SimCoal.
ok
test_PopGen_SimCoal_nodepend ... ok
test_ProtParam ... ok
test_Registry ... ok
test_Restriction ... ERROR
test_SCOP_Astral ... ok
test_SCOP_Cla ... ok
test_SCOP_Des ... ok
test_SCOP_Dom ... ok
test_SCOP_Hie ... ok
test_SCOP_Raf ... ok
test_SCOP_Residues ... ok
test_SCOP_Scop ... ok
test_SProt ... ok
test_SVDSuperimposer ... ok
test_SeqIO ... `^Hok
test_SeqIO_online ... ok
test_SeqUtils ... ok
test_SubsMat ... ok
test_UniGene ... ok
test_Wise ... skipping. Install Wise2 (dnal) if you want to use Bio.Wise.
ok
test_align ... ok
test_docstrings ... ok
test_geo ... ok
test_interpro ... ok
test_kNN ... ok
test_pairwise2 ... ok
test_prodoc ... ok
test_property_manager ... ok
test_prosite ... ok
test_prosite2 ... ok
test_psw ... skipping. Install Wise2 (dnal) if you want to use Bio.Wise.
ok
test_seq ... ok
test_translate ... ok
test_trie ... ERROR
test_triefind ... ERROR

======================================================================
ERROR: test_CAPS
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_tests.py", line 125, in runTest
    self.runSafeTest()
  File "run_tests.py", line 138, in runSafeTest
    cur_test = __import__(self.test_name)
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Tests/test_CAPS.py", line 3, in <module>
    from Bio.Restriction import *
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/Restriction/__init__.py", line 61, in <module>
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/Restriction/Restriction.py", line 96, in <module>
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/Restriction/PrintFormat.py", line 14, in <module>
ImportError: No module named DNAUtils

======================================================================
ERROR: test_KDTree
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_tests.py", line 125, in runTest
    self.runSafeTest()
  File "run_tests.py", line 138, in runSafeTest
    cur_test = __import__(self.test_name)
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Tests/test_KDTree.py", line 1, in <module>
    from Bio.KDTree.KDTree import _neighbor_test, _test
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/KDTree/__init__.py", line 10, in <module>
    #
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/KDTree/KDTree.py", line 13, in <module>
ImportError: cannot import name _CKDTree

======================================================================
ERROR: test_Restriction
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_tests.py", line 125, in runTest
    self.runSafeTest()
  File "run_tests.py", line 138, in runSafeTest
    cur_test = __import__(self.test_name)
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Tests/test_Restriction.py", line 8, in <module>
    from Bio.Restriction import *
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/Restriction/__init__.py", line 61, in <module>
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/Restriction/Restriction.py", line 96, in <module>
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Bio/Restriction/PrintFormat.py", line 13, in <module>
ImportError: cannot import name RanaConfig

======================================================================
ERROR: test_trie
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_tests.py", line 125, in runTest
    self.runSafeTest()
  File "run_tests.py", line 138, in runSafeTest
    cur_test = __import__(self.test_name)
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Tests/test_trie.py", line 6, in <module>
    from Bio import trie
ImportError: cannot import name trie

======================================================================
ERROR: test_triefind
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_tests.py", line 125, in runTest
    self.runSafeTest()
  File "run_tests.py", line 138, in runSafeTest
    cur_test = __import__(self.test_name)
  File "/Users/mh/sagestuff/biopython-1.49b/biopython-1.49b/Tests/test_triefind.py", line 6, in <module>
    from Bio import trie
ImportError: cannot import name trie

======================================================================
FAIL: test_Cluster
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_tests.py", line 125, in runTest
    self.runSafeTest()
  File "run_tests.py", line 162, in runSafeTest
    expected_handle)
  File "run_tests.py", line 263, in compare_output
    % (repr(output_line), repr(expected_line))
AssertionError: 
Output  : 'test_clusterdistance (test_Cluster.TestCluster) ... ERROR\n'
Expected: 'test_clusterdistance (test_Cluster.TestCluster) ... ok\n'

----------------------------------------------------------------------
Ran 95 tests in 147.466s

FAILED (failures=1, errors=5)
You have new mail in /var/mail/was
```



---

archive/issue_comments_027178.json:
```json
{
    "body": "<div id=\"comment:7\" align=\"right\">comment:7</div>\n\nI think most of those errors are being tracked [here](http://bugzilla.open-bio.org).  The KDTree failure might be from it not being built by default.  \n\nI still think this optional package should go out, since it is better than what we had before.  In the now distributed biopython-1.47 there is a lot of stuff that doesn't work because it requires Numeric, and a lot of that now works because of their Numeric to numpy migration.  \n\nOne reason I want this to go in is so I can write better `@`interact examples and I want people to be able to recreate them.  As far as I know I am the only real user of sage+biopython and I want to give it more exposure.  When the final biopython-1.49 comes out I will make a package for it too.\n\nI will start working on a spkg-check, but I would like this to available sooner rather than later.",
    "created_at": "2008-11-29T14:29:38Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/4576#issuecomment-27178",
    "user": "https://github.com/sagetrac-mhampton"
}
```

<div id="comment:7" align="right">comment:7</div>

I think most of those errors are being tracked [here](http://bugzilla.open-bio.org).  The KDTree failure might be from it not being built by default.  

I still think this optional package should go out, since it is better than what we had before.  In the now distributed biopython-1.47 there is a lot of stuff that doesn't work because it requires Numeric, and a lot of that now works because of their Numeric to numpy migration.  

One reason I want this to go in is so I can write better `@`interact examples and I want people to be able to recreate them.  As far as I know I am the only real user of sage+biopython and I want to give it more exposure.  When the final biopython-1.49 comes out I will make a package for it too.

I will start working on a spkg-check, but I would like this to available sooner rather than later.



---

archive/issue_events_051808.json:
```json
{
    "actor": "https://github.com/williamstein",
    "created_at": "2008-11-29T18:54:14Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "label": "https://github.com/sagemath/sage/labels/needs%20work",
    "label_color": "ffff00",
    "label_name": "needs work",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51808"
}
```



---

archive/issue_events_051809.json:
```json
{
    "actor": "https://github.com/williamstein",
    "created_at": "2008-11-29T18:54:14Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "label": "https://github.com/sagemath/sage/labels/positive%20review",
    "label_color": "dfffc0",
    "label_name": "positive review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51809"
}
```



---

archive/issue_comments_027179.json:
```json
{
    "body": "<div id=\"comment:8\" align=\"right\">comment:8</div>\n\nBased on mhamptom's arguments above, and that this is an *optional* package, and he is the official maintainer, I think this should go in. \n\nIt would be nice if at the end of the install a message were printed about \"issues\".",
    "created_at": "2008-11-29T18:54:14Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/4576#issuecomment-27179",
    "user": "https://github.com/williamstein"
}
```

<div id="comment:8" align="right">comment:8</div>

Based on mhamptom's arguments above, and that this is an *optional* package, and he is the official maintainer, I think this should go in. 

It would be nice if at the end of the install a message were printed about "issues".



---

archive/issue_comments_027180.json:
```json
{
    "body": "<div id=\"comment:9\" align=\"right\">comment:9</div>\n\nMerged in the optional spkg repo and mirrored out.\n\nCheers,\n\nMichael",
    "created_at": "2008-11-29T21:43:14Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/4576#issuecomment-27180",
    "user": "https://github.com/sagetrac-mabshoff"
}
```

<div id="comment:9" align="right">comment:9</div>

Merged in the optional spkg repo and mirrored out.

Cheers,

Michael



---

archive/issue_events_051810.json:
```json
{
    "actor": "https://github.com/sagetrac-mabshoff",
    "created_at": "2008-11-29T21:43:14Z",
    "event": "demilestoned",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "milestone_number": null,
    "milestone_title": "sage-3.2.2",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51810"
}
```



---

archive/issue_events_051811.json:
```json
{
    "actor": "https://github.com/sagetrac-mabshoff",
    "created_at": "2008-11-29T21:43:14Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "milestone_number": null,
    "milestone_title": "sage-3.2.1",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51811"
}
```



---

archive/issue_events_051812.json:
```json
{
    "actor": "https://github.com/sagetrac-mabshoff",
    "created_at": "2008-11-29T21:43:14Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "label": "https://github.com/sagemath/sage/labels/positive%20review",
    "label_color": "dfffc0",
    "label_name": "positive review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51812"
}
```



---

archive/issue_events_051813.json:
```json
{
    "actor": "https://github.com/sagetrac-mabshoff",
    "created_at": "2008-11-29T21:43:14Z",
    "event": "closed",
    "issue": "https://github.com/sagemath/sage/issues/4576",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/4576#event-51813"
}
```
